microcosm_pubsub/chain/context_decorators.py

```diff
@@ -0,0 +1,101 @@
+from functools import wraps
+from marshmallow import ValidationError
+from inspect import getfullargspec, ismethod, isfunction
+
+from microcosm_pubsub.chain.decorators import EXTRACTS, BINDS
+
+
+def get_from_context(context, func):
+    """
+    Decorate a function - pass to the function the relevant arguments
+    from a context (dictionary) - based on the function arg names
+
+    """
+    include_self = int(ismethod(func) or not isfunction(func))
+    args_names = getfullargspec(func)[0]
+
+    @wraps(func)
+    def decorate(*passed_args, **passed_kwargs):
+        context_kwargs = {
+            arg_name: context[arg_name]
+            for arg_name in args_names[include_self + len(passed_args):]
+            if arg_name not in passed_kwargs
+        }
+        return func(*passed_args, **passed_kwargs, **context_kwargs)
+    return decorate
+
+
+def save_to_context(context, func):
+    """
+    Decorate a function - save to a context (dictionary) the function return value
+    if the function is marked by @extracts decorator
+
+    """
+    extracts = getattr(func, EXTRACTS, None)
```

Comment:
We have two wrappers:
* `@extracts` - static decorator that know nothing about 'context'
* `save_to_context` - that know the specific 'context'. It get's the details from  `@extracts` decorator.